src/Open3D/GUI/Widget.cpp

```diff
@@ -93,6 +114,10 @@ void Widget::Layout(const Theme& theme) {
 }
 
 Widget::DrawResult Widget::Draw(const DrawContext& context) {
+    if (!impl_->isVisible) {
```

Comment:
Excellent point, individual widgets don't actually check the flag (and like you say, it would be error-prone for them to need to). I committed the fix you suggested.